### ISSUE TITLE
How do I run it in the background?

### ISSUE THREAD

**Issue by semeano** [diegohaz/rest#65]
I have an AWS server set and all is good if I run "npm run prod". But I want the app to keep running in the background, so I can log out and the API keeps running.

Normally I would use [foreverjs](https://github.com/foreverjs/forever), but in this case, reading through package.json, it's easy to see that "npm run prod" will trigger "cross-env NODE_ENV=production nodemon -i \"*.test.js\" -r dotenv-safe/config ."
But when I run this I get an error:

> -bash: /usr/local/bin/cross-env: No such file or directory

Installing the package globally also changed nothing.

So basically my question is, how do run it in the background on my AWS server (with or without forever)? Do I have to build it the old-fashion way?

**Comment by diegohaz** [comment]
`npm run prod` simulates a production environment locally. The command you want is `npm start`.

**Comment by semeano** [comment]
Ok, got it running with "NODE_ENV=prod PORT=8080 forever .", but it keeps the current process on the foreground instead on the background.

As soon as I it Ctrl+C, it stops running.

**Comment by diegohaz** [comment]
Nice! 👍 

I'm not sure if `prod` will work though. You may need to change it to `production`.

**Comment by semeano** [comment]
Nop, all running good. Actually if I run with `production` I get an error:

> /home/ubuntu/obby-api/src/config.js:21
>     throw new Error('You must set the ' + name + ' environment variable');
>     ^
> 
> Error: You must set the MASTER_KEY environment variable
>     at requireProcessEnv (/home/ubuntu/obby-api/src/config.js:8:11)
>     at Object.<anonymous> (/home/ubuntu/obby-api/src/config.js:28:16)
>     at Module._compile (module.js:570:32)
>     at loader (/home/ubuntu/obby-api/node_modules/babel-register/lib/node.js:144:5)
>     at Object.require.extensions.(anonymous function) [as .js] (/home/ubuntu/obby-api/node_modules/babel-register/lib/node.js:154:7)
>     at Module.load (module.js:487:32)
>     at tryModuleLoad (module.js:446:12)
>     at Function.Module._load (module.js:438:3)
>     at Module.require (module.js:497:17)
>     at require (internal/module.js:20:19)

**Comment by diegohaz** [comment]
Yeah, that's the expected behavior when running in production mode.

You should not deploy/version the `.env` file. Instead just set the environment variables on your server (including `MASTER_KEY`).

It works with `prod`, but this way you are not running the app in production mode and things may not be optimized.

**Comment by diegohaz** [comment]
Take a look on how I set the environment variables on this video: https://www.youtube.com/watch?v=6x-ijyG-ack

Or on the deploy section on README: https://github.com/diegohaz/rest#deploy

**Comment by semeano** [comment]
Done! :) Obrigado!
